Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Paged/FarblingProtectionScript.js

```diff
@@ -130,34 +130,45 @@ window.__firefox__.execute(function($) {
         // We need the original length so we can reference it (as we will change it)
         const plugins = window.navigator.plugins
         const originalPluginsLength = plugins.length
-
+        const pluginsPrototype = Object.getPrototypeOf(window.navigator.plugins)
+        
+        const fakePlugins = fakePluginData.map((pluginData) => {
+          return makeFakePlugin(pluginData)
+        })
+        
         // Adds a fake plugin for the given index on fakePluginData
-        const addPluginAtIndex = (newPlugin, index) => {
+        fakePlugins.forEach((newPlugin, index) => {
           const pluginPosition = originalPluginsLength + index
-          window.navigator.plugins[pluginPosition] = newPlugin
-          window.navigator.plugins[newPlugin.name] = newPlugin
-        }
-
-        for (const [index, pluginData] of fakePluginData.entries()) {
-          const newPlugin = makeFakePlugin(pluginData)
-          addPluginAtIndex(newPlugin, index)
-        }
-
-        // Adjust the length of the original plugin array
-        Reflect.defineProperty(window.navigator.plugins, 'length', {
-          value: originalPluginsLength + fakePluginData.length
+          pluginsPrototype[pluginPosition] = newPlugin
+          pluginsPrototype[newPlugin.name] = newPlugin
         })
-
-        // Patch `PluginArray.item(index)` function to return the correct item
-        // otherwise it returns `undefined`
-        const originalItemFunction = plugins.item
-        window.PluginArray.prototype.item = function (index) {
+        
+        // Farble the `item` method on the plugins array
+        const originalItem = window.navigator.plugins.item
+        pluginsPrototype.item = function () {
+          let index = arguments[0]
```

Comment:
Why do `pluginsPrototype.item = function()` + `let index = arguments[0]` instead of `pluginsPrototype.item = function(index)`?

Reply:
Hmm, if new parameters were added, you'd have to change the code to use those anyway, so you could always change the function signature.